contracts/votingMachines/GenesisProtocolLogic.sol

```diff
@@ -0,0 +1,674 @@
+      //If such contract address does not exist in the network (e.g ganache) the contract will use the _stakingToken param as the
+      //staking token address.
+        if (address(GEN_TOKEN_ADDRESS).isContract()) {
+            stakingToken = StandardToken(GEN_TOKEN_ADDRESS);
+        } else {
+            stakingToken = _stakingToken;
+        }
+    }
+
+  /**
+   * @dev Check that the proposal is votable
+   * a proposal is votable if it is in one of the following states:
+   *  PreBoosted,Boosted,QuietEndingPeriod or Queued
+   */
+    modifier votable(bytes32 _proposalId) {
+        require(_isVotable(_proposalId));
+        _;
+    }
+
+    /**
+     * @dev register a new proposal with the given parameters. Every proposal has a unique ID which is being
+     * generated by calculating keccak256 of a incremented counter.
+     * @param _paramsHash parameters hash
+     * @param _proposer address
+     * @param _organization address
+     */
+    function propose(uint , bytes32 _paramsHash,address _proposer,address _organization)
+        external
+        returns(bytes32)
+    {
+      // solium-disable-next-line security/no-block-members
+        require(now > parameters[_paramsHash].activationTime,"not active yet");
+        //Check parameters existence.
+        require(parameters[_paramsHash].queuedVoteRequiredPercentage >= 50);
+        // Generate a unique ID:
+        bytes32 proposalId = keccak256(abi.encodePacked(this, proposalsCnt));
+        proposalsCnt++;
+         // Open proposal:
+        Proposal memory proposal;
+        proposal.callbacks = msg.sender;
+        proposal.organizationId = keccak256(abi.encodePacked(msg.sender,_organization));
+
+        proposal.state = ProposalState.Queued;
+        // solium-disable-next-line security/no-block-members
+        proposal.times[0] = now;//submitted time
+        proposal.currentBoostedVotePeriodLimit = parameters[_paramsHash].boostedVotePeriodLimit;
+        proposal.proposer = _proposer;
+        proposal.winningVote = NO;
+        proposal.paramsHash = _paramsHash;
+        if (organizations[proposal.organizationId] == 0) {
+            if (_organization == address(0)) {
+                organizations[proposal.organizationId] = msg.sender;
+            } else {
+                organizations[proposal.organizationId] = _organization;
+            }
+        }
+        //calc dao bounty
+        uint daoBounty = parameters[_paramsHash].daoBountyConst.mul(averagesDownstakesOfBoosted[proposal.organizationId]);
+        if (daoBounty < parameters[_paramsHash].minimumDaoBounty) {
+            proposal.daoBountyRemain = parameters[_paramsHash].minimumDaoBounty;
+        } else {
+            proposal.daoBountyRemain = daoBounty;
+        }
+        proposal.totalStakes = proposal.daoBountyRemain;
+        proposals[proposalId] = proposal;
+        proposals[proposalId].stakes[NO] = proposal.daoBountyRemain;//dao downstake on the proposal
```

Comment:
If we count DAObounty is a downstake we have to also count it so that the DAO will be able to redeem this. Otherwise, the stake is stuck in the contract.

Reply:
thanks @leviadam 
fix at 
de5481d